### PR TITLE
Use the test harness for the cranelift-tools::filetests test

### PR DESCRIPTION
**Pull request by elliottt**
While working towards running tests with `nextest`, I found that the `filetests` test in `cranelift-tools` doesn't behave in the way that `nextest` expects. Wrapping it in `#[test]` and having it use the test harness fixes this issue.

### CODE REVIEW

**Review by alexcrichton — COMMENTED** [review]
For this particular case could `#[test]` be used? Using libtest-mimic here is basically the same as a single function with `#[test]` I think

**Comment by elliottt** [comment]
> For this particular case could `#[test]` be used? Using libtest-mimic here is basically the same as a single function with `#[test]` I think

Great point, that's totally possible!

File: cranelift/Cargo.toml [hunk 1]
```diff
@@ -19,7 +19,6 @@ path = "src/clif-util.rs"
 [[test]]
 name = "filetests"
 path = "tests/filetests.rs"
-harness = false
```
Comment (alexcrichton) [inline cranelift/Cargo.toml:22]: If you're feeling particularly intrepid, this entire block can be deleted now.

**Review by alexcrichton — APPROVED** [review]
I briefly looked into this awhile ago to see if we could actually hook into libtest-mimic and generate a single test per filetests since I think that'd be preferable here, but `cranelift_filetests` has a fair bit of infrastructure which wasn't trivial to disentangle so I abandoned the effort pretty quickly. This seems like a reasonable compromise though.

File: cranelift/Cargo.toml (hunk 1 above)
Comment (elliottt) [reply cranelift/Cargo.toml:22]: :fire:

**Pull request merged**